docs/client_retries.md

```diff
@@ -0,0 +1,267 @@
+
+Retry Status Codes are configured [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1058-L1082)
+
+Example:
+```java
+      definitions.put(
+          "retry_policy_1_codes",
+          ImmutableSet.copyOf(
+              Lists.<StatusCode.Code>newArrayList(
+                  StatusCode.Code.DEADLINE_EXCEEDED, StatusCode.Code.UNAVAILABLE)));
+```
+
+Retry Params are configured [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1086-L1155)
+
+Example:
+```java
+      settings =
+          RetrySettings.newBuilder()
+              .setInitialRpcTimeout(Duration.ofMillis(60000L))
+              .setRpcTimeoutMultiplier(1.0)
+              .setMaxRpcTimeout(Duration.ofMillis(60000L))
+              .setTotalTimeout(Duration.ofMillis(60000L))
+              .build();
+```
+
+The default configurations are mapping [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1306-L1474)
+
+Example:
+```java
+      builder
+          .exportAssetsSettings()
+          .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("no_retry_0_codes"))
+          .setRetrySettings(RETRY_PARAM_DEFINITIONS.get("no_retry_0_params"));
+```
+
+## Google Cloud Client Library Retry Concepts
+Enabling retries allow an RPC multiple attempts to try and achieve a successful call. A successful call
+is a response from a server that returns an `OK` Status Code (from gRPC) or a `2xx` Status Code (from HttpJson).
+
+### Configurable Retry Params
+- Retry Status Code: Set of failure status codes to retry on
+- Retry Time/ Attempt Bounds: Configurable retry settings (via [RetrySettings](https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.retrying.RetrySettings) class) to define the retry bounds
+
+### When is an RPC retried
+Take a sample RetrySettings configuration
+```java
+settings =
+  RetrySettings.newBuilder()
+      .setInitialRetryDelay(Duration.ofMillis(100L))
+      .setRetryDelayMultiplier(1.3)
+      .setMaxRetryDelay(Duration.ofMillis(60000L))
+      .setInitialRpcTimeout(Duration.ofMillis(60000L))
+      .setRpcTimeoutMultiplier(1.0)
+      .setMaxRpcTimeout(Duration.ofMillis(60000L))
+      .setTotalTimeout(Duration.ofMillis(60000L))
+      .build();
+```
+Individual RPC Bounds (an attempt) are controlled by the following settings:
+- setInitialRetryDelay
+- setRetryDelayMultiplier
+- setMaxRetryDelay
+- setInitialRpcTimeout
+- setRpcTimeoutMultiplier
+- setMaxRpcTimeout
+
+Total RPC Bounds are controlled by the following settings:
```

Comment:
maybe add `Total RPC Bounds (an operation) are controlled` - to differentiate from individual RPC Bounds?

Reply:
Added an explanation for operation/ attempt.